Mathlib/Algebra/Star/Order.lean

```diff
@@ -185,4 +185,85 @@ theorem conjugate_le_conjugate' {a b : R} (hab : a ≤ b) (c : R) : c * a * star
   by simpa only [star_star] using conjugate_le_conjugate hab (star c)
 #align conjugate_le_conjugate' conjugate_le_conjugate'
 
+@[simp]
+lemma star_le_star_iff {x y : R} : star x ≤ star y ↔ x ≤ y := by
+  suffices ∀ x y, x ≤ y → star x ≤ star y from
+    ⟨by simpa only [star_star] using this (star x) (star y), this x y⟩
+  intro x y h
+  rw [StarOrderedRing.le_iff] at h ⊢
+  obtain ⟨d, hd, rfl⟩ := h
+  refine ⟨star d, ?_, star_add _ _⟩
+  induction hd using AddSubmonoid.closure_induction' with
+  | Hs y hy =>
+    obtain ⟨z, rfl⟩ := Set.mem_range.mpr hy
+    rw [star_mul, star_star]
+    exact AddSubmonoid.subset_closure ⟨z, rfl⟩
+  | H1 => rw [star_zero _]; exact zero_mem _
+  | Hmul z _ w _ hz hw => rw [star_add]; exact add_mem hz hw
+
+@[simp]
+lemma star_lt_star_iff {x y : R} : star x < star y ↔ x < y := by
+  by_cases h : x = y
+  · simp [h]
+  · have := star_le_star_iff (x := x) (y := y)
+    rw [le_iff_lt_or_eq, le_iff_lt_or_eq, star_inj] at this
+    simp_all
+
+lemma star_le_iff {x y : R} : star x ≤ y ↔ x ≤ star y := by rw [← star_le_star_iff, star_star]
+
+lemma star_lt_iff {x y : R} : star x < y ↔ x < star y := by rw [← star_lt_star_iff, star_star]
+
+@[simp]
+lemma star_nonneg_iff {x : R} : 0 ≤ star x ↔ 0 ≤ x := by
+  simpa using star_le_star_iff (x := 0) (y := x)
+
+@[simp]
+lemma star_nonpos_iff {x : R} : star x ≤ 0 ↔ x ≤ 0 := by
+  simpa using star_le_star_iff (x := x) (y := 0)
+
+@[simp]
+lemma star_pos_iff {x : R} : 0 < star x ↔ 0 < x := by
+  simpa using star_lt_star_iff (x := 0) (y := x)
+
+@[simp]
+lemma star_neg_iff {x : R} : star x < 0 ↔ x < 0 := by
+  simpa using star_lt_star_iff (x := x) (y := 0)
+
+lemma IsSelfAdjoint.mono {x y : R} (h : x ≤ y) (hx : IsSelfAdjoint x) : IsSelfAdjoint y := by
+  unfold IsSelfAdjoint at *
+  rw [StarOrderedRing.le_iff] at h
+  obtain ⟨d, hd, rfl⟩ := h
+  rw [star_add, hx]
+  induction hd using AddSubmonoid.closure_induction' generalizing x with
+  | Hs _ h => obtain ⟨x, rfl⟩ := h; simp
+  | H1 => simp
+  | Hmul a _ b _ ha hb =>
+    replace ha : star a = a := by simpa using ha (star_zero _)
+    replace hb : star b = b := by simpa using hb (star_zero _)
+    simp only [star_add, ha, hb]
```

Comment:
Again, a bit more conceptual using the fact that `star` is the identity on the range of `s ↦ star s * s`:
```suggestion
  rw [StarOrderedRing.le_iff] at h
  obtain ⟨d, hd, rfl⟩ := h
  rw [IsSelfAdjoint, star_add, hx.star_eq]
  congr
  refine AddMonoidHom.eqOn_closureM (f := starAddEquiv (R := R)) (g := .id R) ?_ hd
  rintro - ⟨s, rfl⟩
  simp
```